crates/matrix-sdk/src/room/timeline/inner.rs

```diff
@@ -747,6 +766,14 @@ impl RoomDataProvider for room::Common {
             }
         }
     }
+
+    async fn push_context(&self) -> Result<Option<PushConditionRoomCtx>> {
+        self.push_context().await
```

Comment:
This looks like endless recursion. I'm guessing this calls an inherent method? Can we remove that one in favor of the trait method so this doesn't look so confusing?

Reply:
This would mean moving the trait outside of the experimental-timeline feature flag, since we use this method outside of the timeline too.

Reply:
Right.. I guess let's keep it as-is then.

Reply:
Actually I was wondering if I should add a single method that returns both the context and the push rules instead of two separate methods, since one is useless without the other. That would solve this naming issue.

Reply:
Fine by me.